src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/hex/Rebar3TreeParser.groovy

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2017 Black Duck Software, Inc.
+ * http://www.blackducksoftware.com/
+ *
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.hub.detect.bomtool.hex
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+import com.blackducksoftware.integration.hub.bdio.graph.MutableDependencyGraph
+import com.blackducksoftware.integration.hub.bdio.graph.MutableMapDependencyGraph
+import com.blackducksoftware.integration.hub.bdio.model.Forge
+import com.blackducksoftware.integration.hub.bdio.model.dependency.Dependency
+import com.blackducksoftware.integration.hub.bdio.model.externalid.ExternalId
+import com.blackducksoftware.integration.hub.bdio.model.externalid.ExternalIdFactory
+import com.blackducksoftware.integration.hub.detect.DetectConfiguration
+import com.blackducksoftware.integration.hub.detect.model.BomToolType
+import com.blackducksoftware.integration.hub.detect.model.DetectCodeLocation
+import com.blackducksoftware.integration.hub.detect.model.DetectProject
+
+import groovy.transform.TypeChecked
+
+@Component
+@TypeChecked
+class Rebar3TreeParser {
+    private final Logger logger = LoggerFactory.getLogger(Rebar3TreeParser.class)
+
+    public static final String LAST_DEPENDENCY_CHARACTER = '└'
+    public static final String NTH_DEPENDENCY_CHARACTER = '├'
+    public static final String HORIZONTAL_SEPARATOR_CHARACTER = '─'
+    public static final String INNER_LEVEL_CHARACTER = '│'
+    public static final String INNER_LEVEL_PREFIX = INNER_LEVEL_CHARACTER + ' '.multiply(2)
+    public static final String OUTER_LEVEL_PREFIX = ' '.multiply(3)
+    public static final String PROJECT_FORGE_STRING = '(project app)'
+
+    @Autowired
+    DetectConfiguration detectConfiguration
```

Comment:
Do you need a configuration here? I'd prefer if we didn't; pass in what you need if you need something? He is big.